browser/static/js/cursor.js

```diff
@@ -0,0 +1,220 @@
+// NOTE: cursor and viewer are more abstract than the brush,
+// how they should relate to each other is an open question.
+// eg, should scale/dimensions be attributes only of viewer? etc
+
+// cursor to handle mouse movement and attributes
+class CalibanCursor {
+  constructor(width, height, scale) {
+    this.width = width
+    this.height = height
+    this.scale = scale;
+    this.pressed = false;
+    this.trace = [];
+
+    // mouse coords
+    // mouse position on canvas, no adjustment for padding
+    this.rawX = 0;
+    this.rawY = 0;
+    // adjusted for padding
+    this.canvasPosX = -5;
+    this.canvasPosY = -5;
```

Comment:
This should probably be an input instead of hardcoded, since we use the `padding` variable in main.